connector/src/main/java/org/geysermc/connector/utils/ChunkUtils.java

```diff
@@ -162,10 +170,21 @@ public static void updateBlock(GeyserSession session, BlockState blockState, Vec
         }
         session.getUpstream().sendPacket(waterPacket);
 
-        // Since Java stores bed colors as part of the namespaced ID and Bedrock stores it as a tag
+        // Since Java stores bed colors/skull information as part of the namespaced ID and Bedrock stores it as a tag
         // This is the only place I could find that interacts with the Java block state and block updates
-        BedBlockEntityTranslator.checkForBedColor(session, blockState, position);
-        SkullBlockEntityTranslator.checkForSkullVariant(session, blockState, position);
+        Reflections ref = new Reflections("org.geysermc.connector.network.translators.block.entity");
+        Set<Class<? extends RequiresBlockState>> classes = ref.getSubTypesOf(RequiresBlockState.class);
+        for (Class<? extends RequiresBlockState> aClass : classes) {
```

Comment:
You should not be doing these reflection lookups every time an update block packet is sent. Instead, iterate through the already present block entities map in `Translators`.